src/main/java/duke/common/Utils.java

```
@@ -0,0 +1,87 @@
+
+    private static final DateTimeFormatter[] POSSIBLE_DATETIME_FORMATS = {
+            DateTimeFormatter.ofPattern("d.M.yyyy HH:mm"),
+            DateTimeFormatter.ofPattern("d-M-yyyy HH:mm"),
+            DateTimeFormatter.ofPattern("d/M/yyyy HH:mm"),
+            DateTimeFormatter.ofPattern("d.M.yyyy HHmm"),
+            DateTimeFormatter.ofPattern("d-M-yyyy HHmm"),
+            DateTimeFormatter.ofPattern("d/M/yyyy HHmm"),
+            DateTimeFormatter.ofPattern("yyyy.M.d HH:mm"),
+            DateTimeFormatter.ofPattern("yyyy-M-d HH:mm"),
+            DateTimeFormatter.ofPattern("yyyy/M/d HH:mm"),
+            DateTimeFormatter.ofPattern("yyyy.M.d HHmm"),
+            DateTimeFormatter.ofPattern("yyyy-M-d HHmm"),
+            DateTimeFormatter.ofPattern("yyyy/M/d HHmm")
+    };
+
+    private static final DateTimeFormatter[] POSSIBLE_DATE_FORMATS = {
+            DateTimeFormatter.BASIC_ISO_DATE,
+            DateTimeFormatter.ofPattern("ddMMyyyy"),
+            DateTimeFormatter.ofPattern("d.M.yyyy"),
+            DateTimeFormatter.ofPattern("d-M-yyyy"),
+            DateTimeFormatter.ofPattern("d/M/yyyy"),
+            DateTimeFormatter.ofPattern("yyyy.M.d"),
+            DateTimeFormatter.ofPattern("yyyy-M-d"),
+            DateTimeFormatter.ofPattern("yyyy/M/d")
+    };
+
+    /**
+     * Parses dateInput into a LocalDateTime object.
+     * Returns null if dateInput cannot be parsed into a LocalDateTime object.
+     *
+     * @param dateInput a string containing the date supplied from the user.
+     * @return a LocalDateTime object or null if failed to parse dateInput.
+     */
+    public static LocalDateTime parseDate(String dateInput) {
+        for (DateTimeFormatter dtf : POSSIBLE_DATE_FORMATS) {
+            try {
+                return LocalDate.parse(dateInput, dtf).atStartOfDay();
+            } catch (DateTimeParseException e) {
+                // Used to catch errors from parsing date.
```

it would be better to throw an exception rather than have an empty catch block

---

Thanks prof for pointing that out!